Dockerfile/base-gcc-clang.Dockerfile

```diff
@@ -0,0 +1,33 @@
+ARG UBUNTU_VERSION
+
+FROM ubuntu:${UBUNTU_VERSION}
+
+ARG GCC_VERSION
+ARG CLANG_VERSION
+
+# Install gcc, clang and basic tools
+RUN    apt-get update \
+        && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+                git \
+                wget \
+                pip \
+                gcc-${GCC_VERSION} \
+                g++-${GCC_VERSION} \
+                clang-${CLANG_VERSION} \
+                libc++-${CLANG_VERSION}-dev \
+                libc++abi-${CLANG_VERSION}-dev \
+                ninja-build \
+                libxml2-dev \
+                libxslt-dev \
+                python3-dev \
+                docker.io \
+        && apt-get clean \
+        && pip install gcovr \
+        && pip install cppyy \
```

Comment:
I think it would be better to install the specific version of `cppyy` and maybe `gcovr`. For `cppyy` the version is set in NWXCmake repo with `NWX_CPPYY_VERSION` environment variable. That said, I'd guess we don't need `cppyy` installation in the base image.

Reply:
I agree. @ryanmrichard Now the settings in NWXCmake do not cover all needed packages, should we prepare a file (maybe in the repo .github?) which contains the version info of every necessary package?

Reply:
@yzhang-23 if it were me, I'd distinguish between organization-wide dependencies (those needed by multiple repos) and repo-specific dependencies (those needed by a single repo, and likely to only ever be needed by a single repo). The former I'd put in a single file. @jwaldrop107 was using NWXCMake for that, but I'm not wed to them living there as opposed to .github. I'm also not wed to the file being written in CMake, *i.e.*, if there's a file format which is easier to use for both the CMaize and CI then go with that. For repo-specific dependencies I'd set the versions in the repo that needs them.

Generally speaking the goal should be decouple things as much as possible, while still maintaining single sources of truth for everything. So if something is needed in multiple places it needs to be factored out. If it's only needed in one place, put it close to that place to avoid coupling.